app/controllers/my_notifications.js

```diff
@@ -1,50 +1,83 @@
-import Ember from 'ember';
-import offers from './offers';
+import Ember from "ember";
+import offers from "./offers";
 
 export default offers.extend({
   sortProperties: ["createdAt:desc"],
   sortedModel: Ember.computed.sort("model", "sortProperties"),
   messagesUtil: Ember.inject.service("messages"),
+  store: Ember.inject.service(),
 
-  allMessages: Ember.computed(function(){
+  allMessages: Ember.computed(function() {
     return this.store.peekAll("message");
   }),
 
-  model: Ember.computed("allMessages.@each.state", "session.currentUser.id", "allMessages.@each.offer.createdBy", function(){
-    var currentUserId = this.get('session.currentUser.id');
+  model: Ember.computed(
+    "allMessages.@each.state",
+    "session.currentUser.id",
+    "allMessages.@each.offer.createdBy",
+    function() {
+      var currentUserId = this.get("session.currentUser.id");
 
-    return this.get("allMessages").rejectBy("state", "never-subscribed").rejectBy("offer.createdBy.id", currentUserId);
-  }),
+      return this.get("allMessages")
+        .rejectBy("state", "never-subscribed")
+        .rejectBy("offer.createdBy.id", currentUserId);
+    }
+  ),
 
+  hasLoadedReadMessages: false,
   showUnread: Ember.computed({
     get: function() {
-      return false;
+      return true;
     },
     set: function(key, value) {
       return value;
     }
   }),
 
-  myNotifications: Ember.computed('showUnread', 'allNotifications', function(){
-    return this.get('showUnread') ? this.get('unreadNotifications') : this.get('allNotifications');
+  myNotifications: Ember.computed("showUnread", "allNotifications", function() {
+    if (this.get("showUnread")) {
+      return this.get("unreadNotifications");
+    }
+    return this.get("allNotifications");
+  }),
+
+  unreadNotifications: Ember.computed("allNotifications.[]", function() {
+    return this.get("allNotifications").rejectBy("unreadCount", 0);
   }),
 
-  unreadNotifications: Ember.computed('allNotifications.[]', function(){
-    return this.get('allNotifications').rejectBy('unreadCount', 0);
+  readNotifications: Ember.computed("allNotifications.[]", function() {
+    return this.get("allNotifications").filterBy("unreadCount", 0);
   }),
 
-  allNotifications: Ember.computed("model.@each.state", function(){
+  allNotifications: Ember.computed("model.@each.state", function() {
     var keys = {};
     var res = [];
-    this.get("sortedModel").forEach(function(message) {
+    this.get("sortedModel").forEach(message => {
       var isPrivate = message.get("isPrivate");
-      var key = isPrivate + message.get("offer.id") + (message.get("item.id") || "");
+      var key =
+        isPrivate + message.get("offer.id") + (message.get("itemId") || "");
       if (!keys[key]) {
-        var props = ["id", "item", "offer", "sender", "createdAt", "isPrivate"];
+        var props = [
+          "id",
+          "itemId",
```

Comment:
notifications are now created with only the itemId, since the items are not loaded on startup anymore